test/models/content_view_test.rb

```diff
@@ -296,6 +296,40 @@ def test_check_ready_to_destroy!
       assert view.check_ready_to_destroy!
     end
 
+    def test_check_ready_to_publish!
+      composite = ContentView.find(katello_content_views(:composite_view))
+      # version with no envs
+      library_view_version_1 = ContentViewVersion.find(katello_content_view_versions(:library_view_version_1))
+      # version in library & dev
+      library_dev_view_version = ContentViewVersion.find(katello_content_view_versions(:library_dev_view_version))
+
+      # tests where the component view versions must exist in target environment
+      restrict_composite_view = Setting.create(:name => 'restrict_composite_view', :category => 'Setting::Katello',
+                                               :settings_type => 'boolean', :default => true)
+
+      composite.components = [library_view_version_1]
+      composite.save!
+      assert_raises RuntimeError do
+        composite.check_ready_to_publish!
+      end
+
+      composite.components = [library_dev_view_version]
+      composite.save!
+      assert composite.check_ready_to_publish!
+
+      # tests where the component view versions do not need to exist in target environment
+      restrict_composite_view.default = false
+      restrict_composite_view.save!
+
+      composite.components = [library_view_version_1]
+      composite.save!
+      assert composite.check_ready_to_publish!
+
+      composite.components = [library_dev_view_version]
+      composite.save!
+      assert composite.check_ready_to_publish!
```

Comment:
This test appears to be covering multiple conditions which we would normally split up into individual test functions isolated to those scenarios.